app/src/main/java/com/tolkiana/tacoapp/TacoFactFragment.kt

```diff
@@ -0,0 +1,40 @@
+package com.tolkiana.tacoapp
+
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import kotlinx.android.synthetic.main.fragment_taco_fact.*
+
+/**
+ * A simple [Fragment] subclass.
+ */
+class TacoFactFragment : Fragment() {
+
+    companion object {
+        private const val ARG_TACO_FACT= "tacoFact"
+
+        fun newInstance(tacoFact: TacoFact): TacoFactFragment{
+            val arguments = Bundle()
+            arguments.putSerializable(ARG_TACO_FACT, tacoFact)
```

Comment:
You can put the tacoFact as a parcelable (which is way more performant than the serializable) using `arguments.putParcelable()`. That would also allow you to remove the `Serializable` implementation in the TacoFact class.

Reply:
You would also need to update the `getSerializable` down below in the `onViewCreated()` method.